pkg/syncer/volume_health_reconciler.go

```diff
@@ -404,9 +406,15 @@ func (rc *volumeHealthReconciler) updateTKGPVC(ctx context.Context, svcPVC *v1.P
 
 	// Check if annotation is the same on PVC in Tanzu Kubernetes Grid and Supervisor Cluster and copy from Supervisor Cluster if different
 	var tkgAnnValue, svcAnnValue string
-	tkgAnnValue, tkgFound := tkgPVCObj.ObjectMeta.Annotations[annVolumeHealth]
-	svcAnnValue, svcFound := svcPVC.ObjectMeta.Annotations[annVolumeHealth]
-	if !tkgFound && svcFound || tkgFound && svcFound && tkgAnnValue != svcAnnValue {
+	var tkgAnnFound, svcAnnFound bool
+	tkgAnnValue, tkgAnnFound = tkgPVCObj.ObjectMeta.Annotations[annVolumeHealth]
+	if svcPVC != nil {
+		svcAnnValue, svcAnnFound = svcPVC.ObjectMeta.Annotations[annVolumeHealth]
+	} else {
+		svcAnnValue = volHealthStatusInAccessible
+	}
+
+	if !tkgAnnFound && svcAnnFound || tkgAnnFound && svcAnnFound && tkgAnnValue != svcAnnValue || svcPVC == nil {
 		log.Infof("updateTKGPVC: Detected volume health annotation change. Need to update Tanzu Kubernetes Grid PVC %s/%s. Existing TKG PVC annotation: %s. New annotation: %s", tkgPVCObj.Namespace, tkgPVCObj.Name, tkgAnnValue, svcAnnValue)
 		tkgPVCClone := tkgPVCObj.DeepCopy()
```

Comment:
Not your code change - why do we have to do a deep copy?

Reply:
Not sure, i couldn't find any previous comment or review about the deep copy